js/ooyala_ssai.js

```diff
@@ -342,12 +347,15 @@ OO.Ads.manager(function(_, $)
      */
     this.resumeAd = function(ad)
     {
-      if (ad && ad.ad && ad.ad.data && this.adIdDictionary[ad.ad.data.id] && currentId3Object) {
-        //Setting the ad callback again since ad was resumed
-        this.adIdDictionary[ad.ad.data.id].adTimer = _.delay(
-          _adEndedCallback(null, ad.ad.data.id),
-          currentId3Object.duration * 1000
-        );
+      if (adMode) {
+        _handleTrackingUrls(this.currentAd, ["resume"]);
+        if (ad && ad.ad && ad.ad.data && this.adIdDictionary[ad.ad.data.id] && _.isNumber(ad.duration)) {
```

Comment:
I think `_.isNumber(ad.duration)` might return true for `NaN`. We could add a a check for `_.isFinite()` if that's the case.

Reply:
@jpfranco changed